hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metrics/HoodieMetrics.java

```diff
@@ -130,6 +140,13 @@ public Timer.Context getIndexCtx() {
     return indexTimer == null ? null : indexTimer.time();
   }
 
+  public Timer.Context getConflictResolutionCtx() {
+    if (config.isMetricsOn() && conflictResolutionTimer == null) {
```

Comment:
shouldn't we check for LockMetricsOn() as well ?

Reply:
Going forward with the infer change I am only checking for LockMetricsOn